src/backend/commands/self/ammo.cpp

```diff
@@ -0,0 +1,131 @@
+			WEAPON::GET_MAX_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_PISTOL_INCENDIARY"), &AMMO_PISTOL_INCENDIARY);
+			WEAPON::SET_PED_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_PISTOL_INCENDIARY"), AMMO_PISTOL_INCENDIARY);
+			WEAPON::GET_MAX_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_PISTOL_TRACER"), &AMMO_PISTOL_TRACER);
+			WEAPON::SET_PED_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_PISTOL_TRACER"), AMMO_PISTOL_TRACER);
+			WEAPON::GET_MAX_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_MG_ARMORPIERCING"), &AMMO_MG_ARMORPIERCING);
+			WEAPON::SET_PED_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_MG_ARMORPIERCING"), AMMO_MG_ARMORPIERCING);
+			WEAPON::GET_MAX_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_MG_FMJ"), &AMMO_MG_FMJ);
+			WEAPON::SET_PED_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_MG_FMJ"), AMMO_MG_FMJ);
+			WEAPON::GET_MAX_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_MG_INCENDIARY"), &AMMO_MG_INCENDIARY);
+			WEAPON::SET_PED_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_MG_INCENDIARY"), AMMO_MG_INCENDIARY);
+			WEAPON::GET_MAX_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_MG_TRACER"), &AMMO_MG_TRACER);
+			WEAPON::SET_PED_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_MG_TRACER"), AMMO_MG_TRACER);
+			WEAPON::GET_MAX_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_SNIPER_EXPLOSIVE"), &AMMO_SNIPER_EXPLOSIVE);
+			WEAPON::SET_PED_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_SNIPER_EXPLOSIVE"), AMMO_SNIPER_EXPLOSIVE);
+			WEAPON::GET_MAX_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_SMG_FMJ"), &AMMO_SMG_FMJ);
+			WEAPON::SET_PED_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_SMG_FMJ"), AMMO_SMG_FMJ);
+			WEAPON::GET_MAX_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_SMG_HOLLOWPOINT"), &AMMO_SMG_HOLLOWPOINT);
+			WEAPON::SET_PED_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_SMG_HOLLOWPOINT"), AMMO_SMG_HOLLOWPOINT);
+			WEAPON::GET_MAX_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_SMG_INCENDIARY"), &AMMO_SMG_INCENDIARY);
+			WEAPON::SET_PED_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_SMG_INCENDIARY"), AMMO_SMG_INCENDIARY);
+			WEAPON::GET_MAX_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_SMG_TRACER"), &AMMO_SMG_TRACER);
+			WEAPON::SET_PED_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_SMG_TRACER"), AMMO_SMG_TRACER);
+			WEAPON::GET_MAX_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_FLAREGUN"), &AMMO_FLAREGUN);
+			WEAPON::SET_PED_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_FLAREGUN"), AMMO_FLAREGUN);
+			WEAPON::GET_MAX_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_FIREWORK"), &AMMO_FIREWORK);
+			WEAPON::SET_PED_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_FIREWORK"), AMMO_FIREWORK);
+			WEAPON::GET_MAX_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_RAILGUN"), &AMMO_RAILGUN);
+			WEAPON::SET_PED_AMMO_BY_TYPE(self::ped, MISC::GET_HASH_KEY("AMMO_RAILGUN"), AMMO_RAILGUN);
+		}
+	};
```

Comment:
this whole method is pointless and just takes up too much space. Another drawback is that if R* added a new gun that takes a different ammo type then this would have to be updated so let's avoid that...